components/calendar/src/types.rs

```diff
@@ -19,10 +19,111 @@ use zerovec::ule::AsULE;
 /// Era codes are shared with Temporal, [see Temporal proposal][era-proposal].
 ///
 /// [era-proposal]: https://tc39.es/proposal-intl-era-monthcode/
-#[derive(Copy, Clone, Debug, PartialEq)]
+#[derive(Copy, Clone, Debug, PartialEq, Ord, PartialOrd, Hash, Eq)]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize, databake::Bake))]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
 #[allow(clippy::exhaustive_structs)] // this is a newtype
+#[cfg_attr(feature = "datagen", databake(path = icu_calendar::types))]
+#[repr(C, packed)]
 pub struct Era(pub TinyStr16);
 
+impl AsULE for Era {
+    type ULE = <TinyStr16 as AsULE>::ULE;
+
+    fn from_unaligned(unaligned: Self::ULE) -> Self {
+        Self(Self::ULE::from_unaligned(unaligned))
+    }
+
+    fn to_unaligned(self) -> Self::ULE {
+        Self::ULE::to_unaligned(self.0)
+    }
+}
+
+#[allow(missing_docs)]
+impl Era {
+    /// Singleton Buddhist era
+    pub const BE: Era = Era(tinystr!(16, "be"));
+    /// Anno Diocletian/After Diocletian (Coptic)
+    pub const AD: Era = Era(tinystr!(16, "ad"));
+    /// Before Diocletian (Coptic)
+    pub const BD: Era = Era(tinystr!(16, "bd"));
+    /// Singleton Chinese era
+    pub const CHINESE: Era = Era(tinystr!(16, "chinese"));
+    /// Singleton Dangi era
+    pub const DANGI: Era = Era(tinystr!(16, "dangi"));
+    /// Incarnation (Ethiopian)
+    pub const INCARNATION: Era = Era(tinystr!(16, "incar"));
+    /// Pre-Incarnation (Ethiopian)
+    pub const PRE_INCARNATION: Era = Era(tinystr!(16, "pre-incar"));
+    /// Anno Mundi (Ethiopian)
+    pub const MUNDI: Era = Era(tinystr!(16, "mundi"));
+    /// Anno Mundi (Hebrew)
+    pub const AM: Era = Era(tinystr!(16, "am"));
+    /// Before Common Era (Julian, Gregorian)
+    pub const BCE: Era = Era(tinystr!(16, "bce"));
+    /// Common Era (Julian, Gregorian)
+    pub const CE: Era = Era(tinystr!(16, "ce"));
+    /// Anno Hegirae (Hijri, Persian)
+    pub const AH: Era = Era(tinystr!(16, "ah"));
+    /// Saka (Indian)
+    pub const SAKA: Era = Era(tinystr!(16, "saka"));
+    /// ISO-8601 default era
+    pub const DEFAULT: Era = Era(tinystr!(16, "default"));
+    /// Pre-ROC
+    pub const ROC_INVERSE: Era = Era(tinystr!(16, "roc-inverse"));
+    /// ROC
+    pub const ROC: Era = Era(tinystr!(16, "roc"));
+
+    /// Alias for [`Self::AD`]
```

Comment:
The aliases go the other way: the canonical TC39 ones are coptic/coptic-inverse, and BD/AD are aliases of them

https://tc39.es/proposal-intl-era-monthcode/#table-eras

Reply:
I marked everything that wasn't documented but still accepted as an alias.